### DISCUSSION TITLE
Lesson 6: Solution - Brownie testing for reverted transactions does not work with pytest.raises() or brownie.reverts()

### DISCUSSION THREAD

**Discussion by n4n0b1t3** [smartcontractkit/full-blockchain-solidity-course-py#1110] [General]
**The problem:** 
In the end of Lesson 6 we are trying to test for a reverted transaction to make sure the "only owner" modifier works as intended. The check relies on a mechanism where a Virtualmachine error is caught when the transaction reverts and thus proves, that the modifier works.

Until brownie v1.18.1 this mechanism won't work, throws an error, and interrupts the test, see my bug request in the eth-brownie repository here:
https://github.com/eth-brownie/brownie/issues/1441

However, the current Python 3.10.2 will only install Brownie version v1.16.4 and any efforts to force the new version on it will result in another error.

**The solution**
The only (for me) acceptable solution is to set up a virtual environment with a specific Python version, in my case 3.9.10, which will install the newest brownie version.

This process is described in this post here:
https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1109

In order to test if your setup has the same issue, you can download the test sources here: 
https://github.com/n4n0b1t3/brownie.reverts.example

**Comment by cromewar** [comment]
Pinning this ;) 

Wil try using your source code.

Nice contribution @n4n0b1t3

**Reply by n4n0b1t3** [reply]
Awesome!
